processing/src/main/java/org/apache/druid/math/expr/ExprEval.java

```diff
@@ -644,6 +644,37 @@ public static Number computeNumber(@Nullable String value)
     return rv;
   }
 
+  /**
+   * Cast an {@link ExprEval} to some {@link ExpressionType} that the value will be compared with. If the value is not
+   * appropriate to use for comparison after casting, this method returns null. For example, the
+   * {@link ExpressionType#DOUBLE} value 1.1 when cast to {@link ExpressionType#LONG} becomes 1L, which is no longer
+   * appropriate to use for equality or range comparisons, while 1.0 is valid.
```

Comment:
Why "or range comparisons"? The value `1.1` is appropriate to use for `long` typed ranges, because, for example, `2L` is greater than `1.1`. The open/closedness of the bounds may need to be adjusted but that does not mean the value is inappropriate.

Reply:
oops, i forgot to update the javadoc since range doesn't use this anymore